ads/google/a4a/utils.js

```diff
@@ -121,10 +122,18 @@ export function googleAdUrl(
     const viewportRect = viewport.getRect();
     const iframeDepth = iframeNestingDepth(win);
     const viewportSize = viewport.getSize();
-    if (ValidAdContainerTypes.indexOf(adElement.parentElement.tagName) >= 0) {
-      queryParams.push({name: 'amp_ct',
-                        value: adElement.parentElement.tagName});
+    // Detect container types.
+    let parentElement = adElement.parentElement;
+    const containerTypeSet = {};
+    while (parentElement) {
```

Comment:
Let's add a count limit just in case (e.g. max 20 iterations).  May be overly cautious but want to make sure we never get a while(true) situation